tests/nvmeof/test_ceph_nvmeof_ns_masking.py

```diff
@@ -0,0 +1,410 @@
+import json
+from copy import deepcopy
+
+from ceph.ceph import Ceph
+from ceph.ceph_admin.common import fetch_method
+from ceph.nvmegw_cli import NVMeGWCLI
+from ceph.nvmeof.initiator import Initiator
+from ceph.utils import get_node_by_id
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+from utility.utils import generate_unique_id
+
+LOG = Log(__name__)
+
+
+def validate_namespace_masking(
+    nsid,
+    subnqn,
+    namespaces_sub,
+    hostnqn_dict,
+    ns_visibility,
+    command,
+    expected_visibility,
+):
+    """Validate that the namespace visibility is correct."""
+
+    if command == "add_host":
```

Comment:
we must start using ha.py for more flexibility of executing anything w.r.t GWs, clients and others. lets avoid NVMeGWCLI  usage directly unless needed.

Reply:
Made changes